jupyterhub/app.py

```diff
@@ -3555,16 +3557,17 @@ async def start(self):
         # Check services health
         self._check_services_health_callback = None
         if self.service_check_interval:
-            self._check_services_health_callback = PeriodicCallback(
```

Comment:
This callback is never set now i think, a mistake?

Reply:
No, these handles only existed for cleanup in `stop`, but then needed to be tracked by name in both start and stop. Putting them in a dictionary means stop doesn't need to be updated when these are added, removed, or changed.

Reply:
![Screenshot_20240129-143135](https://github.com/jupyterhub/jupyterhub/assets/3837114/77bbee8f-33b3-4312-aa14-81fda13d4dbb)

I think maybe the assignment to None should be cleaned up as well? Reviewing via mobile, a bit limited to looking this up thoroughly

Reply:
Good catch! Removed that as well.